Sources/Commands/SwiftPackageTool.swift

```diff
@@ -310,25 +321,32 @@ public class SwiftPackageTool: SwiftTool<PackageToolOptions> {
                     isCodeCoverageEnabled: $2)
                 $0.outputPath = $3?.path
             })
-        
-        let generateCompletionScript = parser.add(
-            subparser: PackageMode.generateCompletionScript.rawValue,
-            overview: "Generate completion script (Bash or ZSH)")
+
+        let completionToolParser = parser.add(subparser: PackageMode.completionTool.rawValue, overview: "Completion tool (for shell completions)")
         binder.bind(
-            positional: generateCompletionScript.add(
-                positional: "flavor", kind: Shell.self,
+            option: completionToolParser.add(
+                option: "--generate-shell-script", kind: Shell.self,
                 usage: "Shell flavor (bash or zsh)"),
-            to: {
-                $0.shell = $1
-            })
+            to: { $0.completionToolMode = .generateShellScript($1) })
+        binder.bind(
+            option: completionToolParser.add(
+                option: "--list-dependencies", kind: Bool.self,
+                usage: "List all dependencies"),
+            to: { $0.completionToolMode = .listDependencies; _ = $1 })
+        binder.bind(
+            option: completionToolParser.add(
+                option: "--list-executables", kind: Bool.self,
+                usage: "List all executables"),
+            to: { $0.completionToolMode = .listExecutables; _ = $1 })
```

Comment:
Is there a cleaner way to say we're only interested in `$0` without explicitly naming them?